src/compiler/checker.ts

```diff
@@ -22978,6 +22978,18 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
             type;
     }
 
+    // This is the same as getBaseTypeOfLiteralType, but checks EnumLiteral last so we get
+    // a specific literal's base type rather than the enum's base type.
+    function getBaseTypeOfLiteralTypeForComparison(type: Type): Type {

```

Comment:
This is a random name picked out of a hat; I didn't know what to call it but I couldn't seem to find an existing function which behaved like this. I'm sure someone knows if there's already a function for this.